service/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateAksStep.java

```diff
@@ -119,6 +71,127 @@ protected void createResource(FlightContext context, ArmManagers armManagers) {
+              .withAgentPoolMode(AgentPoolMode.SYSTEM)
+              .withVirtualNetwork(vNetId, CromwellBaseResourcesFactory.Subnet.AKS_SUBNET.name());
+
+      if (Boolean.parseBoolean(
+          parametersResolver.getValue(
+              CromwellBaseResourcesFactory.ParametersNames.AKS_AUTOSCALING_ENABLED.name()))) {
+        int min =
+            Integer.parseInt(
+                parametersResolver.getValue(
+                    CromwellBaseResourcesFactory.ParametersNames.AKS_AUTOSCALING_MIN.name()));
+        int max =
+            Integer.parseInt(
+                parametersResolver.getValue(
+                    CromwellBaseResourcesFactory.ParametersNames.AKS_AUTOSCALING_MAX.name()));
+        aksPartial = aksPartial.withAutoScaling(min, max);
+      }
+
+      aks =
+          aksPartial
+              .attach()
+              .withDnsPrefix(resourceNameProvider.getName(getResourceType() + DNS_SUFFIX_KEY))
+              .withTags(
+                  Map.of(
+                      LandingZoneTagKeys.LANDING_ZONE_ID.toString(),
+                      landingZoneId.toString(),
+                      LandingZoneTagKeys.LANDING_ZONE_PURPOSE.toString(),
+                      ResourcePurpose.SHARED_RESOURCE.toString()))
+              .create();
+    } catch (ManagementException e) {
+      if (e.getResponse() != null
+          && HttpStatus.CONFLICT.value() == e.getResponse().getStatusCode()) {
+        aks = handleConflictAndMaybeGetAks(context, aksName, e);
+      } else {
+        throw e;
+      }
+    }
+    return aks;
+  }
+
+  private KubernetesCluster handleConflictAndMaybeGetAks(
+      FlightContext context, String aksName, ManagementException e) {
+    return switch (e.getValue().getCode().toLowerCase()) {
+        /*duplicate request (Stairway has resumed flight after interruption)
+        but resource is not ready for use and is still being provisioned*/
+      case "operationnotallowed" -> waitAndMaybeGetAksProvisioned(getMRGName(context), aksName);
+        /*duplicate request (Stairway resume flight after interruption), but resource is ready for use*/
+      case "conflict" -> armManagers
+          .azureResourceManager()
+          .kubernetesClusters()
+          .getByResourceGroup(getMRGName(context), aksName);
+      default -> throw e;
+    };
+  }
+
+  private KubernetesCluster waitAndMaybeGetAksProvisioned(String mrgName, String aksName) {
+    final int pollCycleNumberMax = 30;
+    final int sleepDurationSeconds = 30;
+    int pollCycleNumber = 0;
+    KubernetesCluster existingAks;
+    try {
+      do {
+        if (sleepWhilePollingAksStatus) {
+          /*always sleep except during unit testing*/
+          TimeUnit.SECONDS.sleep(sleepDurationSeconds);
+        }
+        existingAks =
+            armManagers
+                .azureResourceManager()
+                .kubernetesClusters()
+                .getByResourceGroup(mrgName, aksName);
+        if (++pollCycleNumber > pollCycleNumberMax) {
```

Comment:
Why do you increment right after line 172? Shouldn't you check if `existingAks` returned something, in which case there is no need to increment and risk throwing the exception?

Reply:
Here we already exceed maximum number of attempts.  But you are right since we already made a call why not to check it.

Reply:
You are right. There is also extra call.